packages/next/src/build/webpack/plugins/define-env-plugin.ts

```diff
@@ -38,6 +42,94 @@ export interface DefineEnvPluginOptions {
   previewModeId: string | undefined
 }
 
+interface DefineEnv {
+  [key: string]:
+    | string
+    | string[]
+    | boolean
+    | undefined
```

Comment:
falsy values will never be possible because of the check at these lines

https://github.com/vercel/next.js/pull/63128/files#diff-28c0329619b90836564a7769c43ff473dbd28e8e31b75e650624da4d06911763R69

https://github.com/vercel/next.js/pull/63128/files#diff-28c0329619b90836564a7769c43ff473dbd28e8e31b75e650624da4d06911763R86